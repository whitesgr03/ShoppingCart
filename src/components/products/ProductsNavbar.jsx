import "../../style/products/productNavBar.css";
import Icon from "@mdi/react";
import { mdiMagnify } from "@mdi/js";

import { useState } from "react";
import { Link, useLocation, useNavigate } from "react-router-dom";

import PropTypes from "prop-types";

const ProductsNavbar = ({ onSearchParams }) => {
	const [active, setActive] = useState(false);
	const [searchBarValue, setSearchBarValue] = useState("");

	const { pathname } = useLocation();
	const navigate = useNavigate();

	const handleAddBorder = () => setActive(true);
	const handleRemoveBorder = () => setActive(false);
	const handleFilterTextChange = e => setSearchBarValue(e.target.value);
	const handleFilterProduct = e => {
		pathname !== "/shop" && navigate("/shop");
		const value = searchBarValue.trim();
		onSearchParams(value && `search=${value}`);
		e.preventDefault();
	};

	return (
		<div className="navigation">
			<ul className="category">
				<li>
					<Link to="/shop">Shop</Link>
				</li>
				<li> / </li>
				<li>
					<Link to="/shop">Category</Link>
				</li>
			</ul>
			<form
				data-testid="searchBar"
				className={`searchBar ${active ? "active-border" : ""}`}
				onSubmit={handleFilterProduct}
			>
				<div className="search-border">
					<input
						type="search"
						name="search"
						placeholder="search..."
						value={searchBarValue}
						onChange={handleFilterTextChange}
						onFocus={handleAddBorder}
						onBlur={handleRemoveBorder}
					/>
				</div>
				<button type="submit">
					<Icon path={mdiMagnify} />
				</button>
			</form>
		</div>
	);
};

ProductsNavbar.propTypes = {
	onSearchParams: PropTypes.func,
};

export default ProductsNavbar;
