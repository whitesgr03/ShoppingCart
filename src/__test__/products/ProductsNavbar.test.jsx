import { render } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";

import ProductsNavbar from "../../components/products/ProductsNavbar";

describe("Renders ProductsNavbar Component", () => {
	it("Should return ProductsNavbar DOM", () => {
		const routes = [
			{
				path: "/",
				element: <ProductsNavbar />,
			},
		];

		const router = createMemoryRouter(routes, {
			initialEntries: ["/"],
		});

		const { container } = render(<RouterProvider router={router} />);

		expect(container).toMatchSnapshot();
	});
});
