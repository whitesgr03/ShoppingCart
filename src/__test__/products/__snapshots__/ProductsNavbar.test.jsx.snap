// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ProductsNavbar Component Should return ProductsNavbar DOM 1`] = `
<div>
  <ul
    class="category"
  >
    <li>
      <a
        href="/shop"
      >
        Shop
      </a>
    </li>
    <li>
       / 
    </li>
    <li>
      <a
        href="/shop"
      >
        Category
      </a>
    </li>
  </ul>
</div>
`;
